src/cmap/worker.rs

```diff
@@ -505,6 +505,14 @@ pub(super) struct PoolWorkerHandle {
     sender: mpsc::Sender<()>,
 }
 
+impl PoolWorkerHandle {
+    #[cfg(test)]
+    pub(super) fn new_mocked() -> Self {
+        let (_, s) = HandleListener::new();
```

Comment:
Seeing this usage of `HandleListener::new()` makes me realize that it's a bit confusing to have a `new` method return a tuple rather than just `Self`. Could we rename that method to be a bit more descriptive?

Reply:
Awesome, thanks